packages/core/src/search/search.ts

```diff
@@ -407,6 +411,28 @@ function addFilter(searchRequest: SearchRequest, filter: Filter): void {
   }
 }
 
+const subexpressionPattern = /{{([^{}]+)}}/g;
+
+/**
+ * Parses an extended FHIR search criteria string (i.e. application/x-fhir-query), evaluating
+ * any embedded FHIRPath subexpressions (e.g. `{{ %patient.id }}`) with the provided variables.
+ *
+ * @see https://hl7.org/fhir/fhir-xquery.html
+ * @param query The X-Fhir-Query string to parse
+ * @param variables Values to pass into embedded FHIRPath expressions
+ * @returns The parsed search request
+ */
+export function parseXFhirQuery(query: string, variables: Record<string, TypedValue>): SearchRequest {
+  query = query.replaceAll(subexpressionPattern, (_, expr) => {
+    const replacement = evalFhirPathTyped(expr, [], variables);
+    if (replacement.length !== 1) {
+      return '';
+    }
+    return formatTypedValue(replacement[0]);
```

Comment:
I'm not quite sure about the context here, but do we need to URL-escape the return value?  i.e. `encodeURIComponent`?

Reply:
My expectation is that the context is defined by the caller: they can encode if adding directly to a URL